substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -1499,6 +1496,17 @@ impl<T: Config> Pallet<T> {
         pricing_policy: &pallet_tfgrid_types::PricingPolicy<T::AccountId>,
         amount: BalanceOf<T>,
     ) -> DispatchResultWithPostInfo {
+        log::info!(
+            "Distributing cultivation rewards for contract {:?} with amount {:?}",
+            contract.contract_id,
+            amount,
+        );
+
+        // If the amount is zero, return
+        if amount == BalanceOf::<T>::saturated_from(0 as u128) {
```

Comment:
If I follow the logic above, if `amount == 0` it is because user `usable_balance == 0`
Wouldn't be better if we handle this case at beginning of `bill_contract()`
Billing a user with no funds doesn't make sense...
Or maybe because `bill_contract()` does a lot more than billing the user 🥲

Reply:
`bill_contract` has to many flows to just return at the start of the function.